debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/SourceInfo.java

```diff
@@ -320,8 +295,8 @@ public void changeStreamEvent(String replicaSetName, ChangeStreamDocument<BsonDo
         long wallTime = 0L;
         if (changeStreamEvent != null) {
             BsonTimestamp ts = changeStreamEvent.getClusterTime();
```

Comment:
Here we get the timestamp from the change stream event. However based on my observation this timestamp usually has a 1 lower increment compared to the timestamp encoded in the resume token's `_data` field. 

Heart events on the other hand are producing the `ts` field which is extracted directly from the resume token.

Does this pose any problems? Technically the timestamp is not used anywhere anymore. Since the TS is part of the offset I assume we cannot remove it as that would change the  schema.

Reply:
Timestamp cab be removed form offsets when not needed if the connector is prepared to next expect it.
It should be kept in `source` info block as `ts_ms` as it represents event time of the event.